### ISSUE TITLE
Select token and certificate using Id or serial number instead of label 

### ISSUE THREAD

**Issue by MhmodTayel** [MatthiasValvekens/pyHanko#149]
When using s PKCS#11 token it's required to pass only the label of token and certificate. I have issue here if there are more than one token have the same label also with certificate if i have more than one certificate with the same label.

A solution for this issue could be using the serial number of the token which is a unique parameter unlike token label also with certificate we can use it's Id or another great solution using certificate URL like p11tool [p11tool](https://man7.org/linux/man-pages/man1/p11tool.1.html)

`pyhanko sign addsig pkcs11 --lib /path/to/module.so \
    --token-serial 83d722f99ffa9e12 --cert-id 0cbaf441eec0586889c721d4591b37b0 document.pdf output.pdf`
   
   or 
   `pyhanko sign addsig pkcs11 --lib /path/to/module.so \
    --cert-url 'pkcs11:model=SoftHSM%20v2;manufacturer=SoftHSM%20project;serial=77120d5dc2acebdb;token=token1;id=%5D%5D%52%B5%C7%73%FF%8D%38%58%E7%C8%4F%46%34%FE;object=cert1;type=cert' document.pdf output.pdf`

**Comment by MatthiasValvekens** [comment]
Hi @MhmodTayel,

You can select a token by slot number using `--slot-no`. For the certificate/key selection, the API already supports querying by ID instead of label, but you're right that this isn't exposed in the CLI right now.

I'll see what I can do. Labelling as `enhancement` for now.

**Comment by MhmodTayel** [comment]
It would be great if I select the token using it's serial number as when using slot number won't be efficient because of switching slots so using serial number will do just fine.
Thank you so much for considering this feature request

**Comment by MatthiasValvekens** [comment]
Hi @MhmodTayel,

I added some plumbing to handle filtering tokens by serial number to the library API and to the config parser. See linked commit above.

Also, it turns out that selecting by key ID or cert ID was also already possible in the CLI, albeit not via command line flags directly: you need to put the parameters in the configuration file instead.

For example, once the linked commit is merged, you could put the following snippet into your config file (by default, `pyhanko.yml`):

```yaml
pkcs11-setups:
  my-favourite-token:
    module-path: /path/to/module.so
    token-criteria:
      serial: 83d722f99ffa9e12
    cert-id: 0cbaf441eec0586889c721d4591b37b0
```

To use these settings, you would then invoke pyhanko as follows:

```bash
pyhanko sign addsig pkcs11 --p11-setup my-favourite-token document.pdf output.pdf
```

I generally try to push advanced options like this to the config file as opposed to the command line itself, since reconciling CLI switches with config is already messy enough as-is.

**Comment by MatthiasValvekens** [comment]
FYI, this was released as part of `0.14.0` yesterday.
